hudi-hadoop-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileUtils.java

```diff
@@ -40,6 +43,39 @@
  * Util class for HFile reading and writing in Hudi
  */
 public class HoodieHFileUtils {
+
+  static class SeekableByteArrayInputStream extends ByteBufferBackedInputStream
```

Comment:
Should this be in an independent file without being nested in `HoodieHFileUtils`?